### PR TITLE
[CI Environment] Added automation to update the Static Validation Tests documentation

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- Added a utility `Set-StaticTestDocumentation` to update the Wiki documentation for static Pester tests
- Added an Azure DevOps and GitHub pipeline to automatically update the Wiki documentation
- Removed redundant test
- Improved description of the tests

| Pipeline |
| - |
| [.Platform: Update Static Test Documentation](https://github.com/Azure/ResourceModules/actions/workflows/platform.updateStaticTestDocs.yml) |
| [Build Status](https://dev.azure.com/carml/CARML/_build/latest?definitionId=129&branchName=users%2Falsehr%2F1428_pesterTests) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: .azuredevops/platformPipelines/platform.updateStaticTestDocs.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name: ".Platform - Update Static Test Documentation"
```
Comment (AlexanderSehr) [inline .azuredevops/platformPipelines/platform.updateStaticTestDocs.yml:1]: Blocking comment until release 0.9

File: .github/workflows/platform.updateStaticTestDocs.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+name: ".Platform: Update Static Test Documentation"
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - 'utilities/pipelines/staticValidation/module.tests.ps1'
```
Comment (eriqua) [inline .github/workflows/platform.updateStaticTestDocs.yml:9]:
```suggestion
      - '.github/workflows/platform.updateStaticTestDocs.yml'
      - 'utilities/pipelines/staticValidation/module.tests.ps1'
      - utilities/tools/platform/Set-StaticTestDocumentation.ps1
```

File: .azuredevops/platformPipelines/platform.updateStaticTestDocs.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+pr: none
+
+trigger:
+  batch: true
+  branches:
+    include:
+      - main
+  paths:
+    include:
+      - 'utilities/pipelines/staticValidation/module.tests.ps1'
```
Comment (eriqua) [inline .azuredevops/platformPipelines/platform.updateStaticTestDocs.yml:12]:
```suggestion
      - .azuredevops/platformPipelines/platform.updateStaticTestDocs.yml
      - 'utilities/pipelines/staticValidation/module.tests.ps1'
      - utilities/tools/platform/Set-StaticTestDocumentation.ps1
```

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**